lib/block.mli

```diff
@@ -34,6 +34,21 @@ val blkgetsize: string -> Unix.file_descr -> [ `Ok of int64 | `Error of error ]
 
 val connect : string -> [`Ok of t | `Error of error] io
 
+type config = {
+  buffered: bool; (** true if I/O hits the OS disk caches, false if "direct" *)
+  sync: bool; (** true if [flush] flushes all caches, including disk drive caches *)
+  path: string; (** path to the underlying file *)
+}
+(** Configuration of a device *)
+
+val connect_uri : Uri.t -> [`Ok of t | `Error of error] io
```

Comment:
I could add another function (`connect_config`?) but in my current use-case (within https://github.com/docker/hyperkit) the main process is written in C and accepts a string as a command-line argument of the form: 
```
%d,virtio-blk,file://%s,format=qcow
```
where the `file://%s` names the file to be opened.

I'd like to avoid changing the C code whenever the OCaml API is expanded here -- if I keep it as  something generic I can get the C code to pass it down into the OCaml code unmodified, so a further expansion or change would only involve changes here.

Another option would be to change `connect` which currently accepts strings of the form:

- `/foo/bar`
- `buffered:/foo/bar`

and add support for

- `file:///foo/bar?sync=1&buffered=0...`

WDYT?

In an ideal world (where hypervisors are written in OCaml :) I'd probably just change `connect` to have a proper typed interface, although that would require changes in many more places.